core/setup.py

```diff
@@ -29,6 +29,34 @@
 description = """With dbt, data analysts and engineers can build analytics \
 the way engineers build applications."""
 
+_install_requires = [
+    "Jinja2==3.1.2",
+    "agate>=1.6,<1.6.4",
+    "click>=7.0,<9",
+    "colorama>=0.3.9,<0.4.6",
+    "hologram>=0.0.14,<=0.0.15",
+    "isodate>=0.6,<0.7",
+    "logbook>=1.5,<1.6",
+    "mashumaro[msgpack]==3.0.4",
+    "minimal-snowplow-tracker==0.0.2",
+    "networkx>=2.3,<2.8.1;python_version<'3.8'",
+    "networkx>=2.3,<3;python_version>='3.8'",
+    "packaging>=20.9,<22.0",
+    "sqlparse>=0.2.3,<0.5",
+    "typing-extensions>=3.7.4",
+    "werkzeug>=1,<3",
+    # the following are all to match snowflake-connector-python
+    "requests<3.0.0",
+    "idna>=2.5,<4",
+    "cffi>=1.9,<2.0.0",
+    "pyyaml>=6.0",
+]
+
+if "DBT_WASM_BUILD" in os.environ and int(os.environ["DBT_WASM_BUILD"]) == 1:
+    # binary dependency not supported in pyodide
+    pass
+else:
+    _install_requires.insert(14, "dbt-extractor~=0.4.1") 
```

Comment:
Is there any cleaner way for us to manage this? Would this add another point of complexity that would prevent us from moving to `pyproject.toml` in the future? (https://github.com/dbt-labs/dbt-core/issues/4674)